### PR TITLE
Plugin: Fix REST API class redeclaration

### PR DESCRIPTION
**Pull request by gziolo**
## What?

When testing the Gutenberg plugin with the most recent version of WordPres core it throws fatal error after: https://github.com/WordPress/wordpress-develop/commit/a807e86391817a45dd82f54bef234be1a28d4c6d. 

## Why?

2 new classes weren't correctly guarded against redeclaration.

## How?

Correctly check whether classes just added to WordPress core are already declared.

## Testing Instructions

All CI checks should pass again.

### CODE REVIEW

**Review by Mamaduka — APPROVED** [review]

**Review by ntsekouras — APPROVED** [review]
Looks good, thanks @gziolo ! Let's wait for the CI first before landing though 😄

**Comment by gziolo** [comment]
The performance tests job failed, but I belive it's because we run it against `trunk` which is invalid at the moment.

**Comment by gziolo** [comment]
Cherry-picked this PR to the Gutenberg 13.0 plugin release to ensure it can be tested with the latest version of WordPress core.

**Pull request merged**
